extensions/spring-xd-extension-http/src/main/java/org/springframework/integration/x/http/NettyHttpInboundChannelAdapter.java

```diff
@@ -198,11 +235,21 @@ protected void doStart() {
 			this.messageConverter = new NettyInboundMessageConverter(getMessageBuilderFactory());
 		}
 		executionHandler = new ExecutionHandler(executor);
-		bootstrap = new ServerBootstrap(new NioServerSocketChannelFactory(Executors.newCachedThreadPool(),
-				Executors.newCachedThreadPool()));
-		bootstrap.setOption("child.tcpNoDelay", true);
-		bootstrap.setPipelineFactory(new PipelineFactory());
-		bootstrap.bind(new InetSocketAddress(this.port));
+		/* begin whu */
+		try {
+			/* end whu */
+			bootstrap = new ServerBootstrap(new NioServerSocketChannelFactory(Executors.newCachedThreadPool(),
+					Executors.newCachedThreadPool()));
+			bootstrap.setOption("child.tcpNoDelay", true);
+			bootstrap.setPipelineFactory(new PipelineFactory());
+			bootstrap.bind(new InetSocketAddress(this.port));
+			/* begin whu */
+		}
+		catch (Exception e) {
+			/* bind exception -> server already created by another channel */
+			bootstrap = null;
```

Comment:
Oh, I see why you're catching it now. Hmmm...

This won't work; consider...

deploy stream a
deploy stream b
undeploy stream a

the bootstrap will be torn down, breaking stream b

I believe we need to store the bootstrap in some shared state and only shut it down when the last module using that port is undeployed.